src/zenml/recipes/stack_recipe_service.py

```diff
@@ -166,3 +179,52 @@ def get_service(cls, recipe_path: str) -> Optional["StackRecipeService"]:
             return None
         except FileNotFoundError:
             return None
+
+    def get_vars(self) -> Dict[str, Any]:
+        """Get variables as a dictionary.
+
+        Returns:
+            A dictionary of variables to use for the stack recipes
+            derived from the tfvars.json file.
+
+        Raises:
+            FileNotFoundError: if the values.tfvars.json file is not
+                found in the stack recipe.
+            TypeError: if the file doesn't contain a dictionary of variables.
+        """
+        vars = super().get_vars()
+        # update zenml version to current version
+        vars[ZENML_VERSION_VARIABLE] = zenml.__version__
+        return vars
+
+    def get_deployment_info(self) -> str:
+        """Return deployment details as a YAML document.
+
+        Returns:
+            A YAML document that can be passed as config to
+            the server deploy function.
+        """
+        provider = yaml_utils.read_yaml(
+            file_path=os.path.join(
+                self.terraform_client.working_dir, "metadata.yaml"
+            )
+        )["Cloud"]
+
+        config = {
+            "name": f"{provider}",
+            "provider": f"{provider}",
+            "username": "default",
+            "password": "zenml",
```

Comment:
Just an idea: if you leave out the username and password, the user will be prompted to enter them by the `zenml deploy` CLI command execution. Isn't that better than using default values here ?

Reply:
that's certainly better. Thanks for the idea!